src/main/scala/battery/bot/database/PersistenceService.scala

```diff
@@ -1,22 +1,27 @@
 package battery.bot.database
 
+import battery.bot.database.DevicesQueries.insertDevices
 import cats.effect.IO
 import battery.bot.database.PricesQueries.insertPrices
+import battery.bot.database.UsersQueries.insertUsers
 import doobie.Transactor
 import doobie.implicits._
 
+import java.util.{Random, UUID}
 import java.time.Instant
 
 class PersistenceService(ta: Transactor[IO]) {
 
-  def addPrice(hour: Int, price: BigDecimal) ={
+  def addPrice(hour: Int, price: BigDecimal): IO[Int] ={
     insertPrices(Instant.now(), price).run.transact(ta)
```

Comment:
This should build the instant based on the current date and the hour that you are receiving